### PR TITLE
Memoize parse_fontconfig_pattern; speeds up test suite by ~1min.

### PR DESCRIPTION
**Pull request by anntzer**
On my laptop this patch drops the duration of the test suite from 604s
to 554s (with the inkscape patch on as well).

(The fontconfig syntax is simple enough that one could also probably write a simple regex parser.)

attn @NelleV

### CODE REVIEW

File: lib/matplotlib/fontconfig_pattern.py [hunk 1]
```diff
@@ -166,7 +172,11 @@ def _property(self, s, loc, tokens):
             self._properties.setdefault(key, []).extend(val)
         return []
 
-parse_fontconfig_pattern = FontconfigPatternParser().parse
+# `parse_fontconfig_pattern` is a bottleneck during the tests because it is
+# repeatedly called when the rcParams are reset (to validate the default
+# fonts).  In practice, the cache size doesn't grow beyond a few dozen entries
+# during the test suite.
+parse_fontconfig_pattern = lru_cache()(FontconfigPatternParser().parse)
```
Comment (QuLogic) [inline lib/matplotlib/fontconfig_pattern.py]: Should be another blank line above the function.

Comment (anntzer) [inline lib/matplotlib/fontconfig_pattern.py]: done

**Comment by anntzer** [comment]
For the record I have also tried updating the cache on mathtext.parse to use a LRU cache (the current `maxdict` is a bit silly...).  Even though that function seems to be another bottleneck currently (~12% total test time with the inkscape patch, without this patch), using a large cache does not appear to help whatsoever (whether with a max cache size of 256 or 1024, I get 1705 hits and 1659 misses, and the runtime is essentially the same).

**Comment by NelleV** [comment]
Thanks @anntzer !

**Pull request merged**